### PR TITLE
EPMDJ-6255 Selenium tests must be thread safe

### PR DESCRIPTION
**Pull request by Mazel-Tovr**

### CODE REVIEW

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 1]
```diff
@@ -40,12 +33,13 @@ actual object TestListener {
                 TestInfo::name.name to test,
                 TestInfo::startedAt.name to System.currentTimeMillis()
             )
```
Comment (LikhanovMaxim) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:35]: logger.info { "Test: $it STARTED" }
why did you remove the debug log?

**Review by LikhanovMaxim — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 2]
```diff
@@ -17,17 +18,9 @@ actual object TestListener {
...
-                    testProperties.put(testId, persistentHashMapOf(paramName to value))
-                }
-            } else {
-                _testInfo.getAndUpdate { testProperties ->
-                    testProperties.put(
-                        testId,
-                        testProperties[testId]?.put(paramName, value)
-                    )
-                }
+            _testInfo.update { testProperties ->
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:21]: `_testInfo.update { testProperties ->
            val currentInfo = testProperties[testId] ?: persistentHashMapOf()
            testProperties.put(testId, currentInfo + vals)
 }`

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt [hunk 3]
```diff
@@ -40,12 +33,13 @@ actual object TestListener {
                 TestInfo::name.name to test,
                 TestInfo::startedAt.name to System.currentTimeMillis()
             )
+            DevToolsClientThreadStorage.getDevTool()
+                ?.addHeaders(mapOf(TEST_NAME_HEADER to it, SESSION_ID_HEADER to (ThreadStorage.sessionId() ?: "")))
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:37]: formatting

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/http/selenium/ChromeDevTool.kt [hunk 4]
```diff
@@ -16,35 +16,46 @@ private const val DEV_TOOL_PROPERTY_NAME = "webSocketDebuggerUrl"
 
 object DevToolsClientThreadStorage {
     private val logger = Logging.logger(ChromeDevTool::class.java.name)
-    internal var crhmT: InheritableThreadLocal<ChromeDevTool> = InheritableThreadLocal()
+    private val crhmT: InheritableThreadLocal<ChromeDevTool> = InheritableThreadLocal()
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/http/selenium/ChromeDevTool.kt:19]: rename this

**Review by ViktoriiaLebedeva — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/http/selenium/Selenium.kt [hunk 5]
```diff
@@ -83,7 +78,7 @@ class Selenium : Strategy() {
         )
         ctClass.getDeclaredMethod("get").insertBefore(
             """
-                if ($IF_CONDITION) {
+                if ($IF_CONDITION && !${DevToolsClientThreadStorage::class.java.name}.INSTANCE.${DevToolsClientThreadStorage::isHeadersAdded.name}()) {
```
Comment (ViktoriiaLebedeva) [inline src/runtimeMain/kotlin/com/epam/drill/test/agent/instrumentation/http/selenium/Selenium.kt:81]: take out second condition as with IF_CONDITION

**Review by ViktoriiaLebedeva — CHANGES_REQUESTED** [review]

File: src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt (hunk 1 above)
Comment (Mazel-Tovr) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:35]: we set debug only for that artifact to adopt the project to not to turn on a trace in the agent. I don't think "test start" is debugging info

Comment (LikhanovMaxim) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:34]: 1) we have been decided that change to INFO => logger.info { "Test: $it STARTED" } 
2) also we had a debug log with thread id etc. Why did remove it? It will be helpful when we investigate a bug in next time

Comment (Mazel-Tovr) [reply src/runtimeMain/kotlin/com/epam/drill/test/agent/TestListener.kt:34]: 1. ok 
2. We used that logging to find a case of our problem, when we fixed it, we don't need that log anymore, those logs were only on one specific case, ofc I can back them if necessary, but don't think they can help us

**Review by LikhanovMaxim — APPROVED** [review]

**Review by ViktoriiaLebedeva — APPROVED** [review]
